specs/_features/ePBS/validator.md

```diff
@@ -0,0 +1,243 @@
+    ``assignment`` returned is a tuple of the following form:
+        * ``assignment[0]`` is the list of validators in the committee
+        * ``assignment[1]`` is the slot at which the committee is assigned
+    Return None if no assignment.
+    """
+    next_epoch = Epoch(get_current_epoch(state) + 1)
+    assert epoch <= next_epoch
+
+    start_slot = compute_start_slot_at_epoch(epoch)
+    committee_count_per_slot = get_committee_count_per_slot(state, epoch)
+    for slot in range(start_slot, start_slot + SLOTS_PER_EPOCH):
+        for index in range(committee_count_per_slot):
+            committee = get_ptc_committee(state, Slot(slot))
+            if validator_index in committee:
+                return committee, Slot(slot)
+    return None
+```
+
+### Lookahead
+
+The beacon chain shufflings are designed to provide a minimum of 1 epoch lookahead
+on the validator's upcoming committee assignments for attesting dictated by the shuffling and slot.
+Note that this lookahead does not apply to proposing, which must be checked during the epoch in question.
+
+`get_ptc_committee_assignment` should be called at the start of each epoch
+to get the assignment for the next epoch (`current_epoch + 1`).
+A validator should plan for future assignments by noting their assigned ptc committee
+slot and planning to participate in the ptc committee subnet.
+
+Specifically a validator should:
+* Find peers of the pubsub topic `beacon_attestation_{subnet_id}`.
+    * If the validator is assigned to be an aggregator for the slot (see `is_aggregator()`) [New in MaxEB], then subscribe to the topic.
+
+### Inclusion list proposal
+
+ePBS introduces forward inclusion list. The detail design is described in this [post](https://ethresear.ch/t/no-free-lunch-a-new-inclusion-list-design/16389)
+Proposer is introduced to construct and broadcast `InclusionList` alongside `SignedBeaconBlock`.
+- Proposer for slot `N` submits `SignedBeaconBlock` and in parallel submits `InclusionList` to be included at the beginning of slot `N+1`.
+- Within `InclusionList`, `Transactions` are list of transactions that the proposer wants to include at the beginning of slot `N+1`.
+- Within `inclusionList`, `Summaries` are lists consisting on addresses sending those transactions and their gas limits. The summaries are signed by the proposer.
+- Proposer is allowed to send many of these pairs that aren't committed to its beacon block so no double proposing slashing is involved.
+
+#### Constructing the `InclusionList`
+
+To obtain an inclusion, a block proposer building a block on top of a `state` must take the following actions:
+
+1. Retrieve inclusion list by calling `get_inclusion_list`.
+
+```python
+def get_inclusion_list(payload_id: Optional[PayloadId], execution_engine: ExecutionEngine) -> InclusionListResponse:
+        return execution_engine.get_payload(payload_id).inclusion_list
+```
+
+2. Build `InclusionListSummary`
```

Comment:
I think the EL should return this and this should be part of the `InclusionListResponse`  class, the validator should simply sign it. Having the validator build the summary would imply the validator parsing the transactions slice, which we want to keep as opaque bytes in the CL